templates/types/streaming/fastapi/app/engine/generate.py

```diff
@@ -54,14 +57,24 @@ def generate_datasource():
     ingestion_pipeline.vector_store = vector_store
 
     # Run the ingestion pipeline and store the results
-    ingestion_pipeline.run(show_progress=True, documents=documents)
+    nodes = ingestion_pipeline.run(show_progress=True, documents=documents)
 
-    # Default vector store only keeps data in memory, so we need to persist it
-    # Can remove if using a different vector store
+    # SimpleVectorStore keeps the data in memory only - needs to be persisted explicitly
```

Comment:
i think we should put this whole code in its own `persist` function

Reply:
Yes, i also think that we should move the persist to another function.